app/src/main/java/ch/epfl/qedit/view/question/QuestionFragment.java

```diff
@@ -34,17 +37,38 @@ public View onCreateView(
 
         View view = inflater.inflate(R.layout.question_fragment, container, false);
 
-        final TextView questionTitle = (TextView) view.findViewById(R.id.question_title);
-        questionTitle.setText((question.getIndex() + 1) + ") " + question.getTitle());
+        final TextView questionTitle = view.findViewById(R.id.question_title);
+        String questionTitleStr = (question.getIndex() + 1) + ") " + question.getTitle();
+        questionTitle.setText(questionTitleStr);
 
-        final TextView questionDisplay = (TextView) view.findViewById(R.id.question_display);
+        final TextView questionDisplay = view.findViewById(R.id.question_display);
         questionDisplay.setText(question.getText());
 
+        // Start a new AnswerFragment with the correct AnswerFormat
+        Bundle bundle = new Bundle();
+        bundle.putSerializable("format", answerFormat);
+        AnswerFragment answerFragment = new AnswerFragment();
+        answerFragment.setArguments(bundle);
+
+        FragmentTransaction fragmentTransaction =
+                Objects.requireNonNull(getActivity())
+                        .getSupportFragmentManager()
+                        .beginTransaction();
+        fragmentTransaction.replace(R.id.answer_fragment_container, answerFragment).commit();
+
         return view;
     }
 
     @Override
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
     }
+
+    // Concrete Visitor
+    private class Visitor implements AnswerFormat.Visitor {
+        @Override
+        public void visitNumberField(AnswerFormat.NumberField field) {
+            answerFormat = field;
```

Comment:
I don't really like this way of doing it but I think that the accept() method normally returns void so I kind of had to add a private field and do it in this way.